data/effects/swirl.effect

```diff
@@ -0,0 +1,144 @@
+    float minimum = 0.0;
+    float maximum = 10.0;
+> = 1.0;
+
+uniform float aspect_ratio<
+    string name = "Aspect Ratio";
+    string description = "Adjusts the aspect ratio for the associated distortion.";
+    string field_type = "slider";
+    float step = 0.01;
+    float minimum = -1;
+    float maximum = 1;
+> = 0.0;
+
+uniform bool animate<
+    string name = "Animate";
+    string description = "Animates the effect, making it twist back and forth.";
+> = false;
+
+uniform bool inverse_angle<
+    string name = "Use Inverse Angle";
+    string description = "Inverts the angle, making the edges more distorted than the center.";
+> = false;
+
+//-----------------------------------------------------------------------------
+// Textures
+//-----------------------------------------------------------------------------
+uniform texture2d swirl_result;
+
+//-----------------------------------------------------------------------------
+// Samplers
+//-----------------------------------------------------------------------------
+sampler_state texture_sampler {
+	Filter    = Linear;
+	AddressU  = Mirror;
+	AddressV  = Mirror;
+};
+
+//-----------------------------------------------------------------------------
+// Functions
+//-----------------------------------------------------------------------------
+
+float2 swirl_transform (float percent, float2 input_vertex, float2 effect_center) {
+    float theta = percent * percent * radians(angle * (animate == true ? sin(Time.x) : 1.0) );
+    float s =  sin(theta);
+    float c =  cos(theta);
+    float2x2 transform = float2x2(
+        c, -s,
+        s, c
+    );
+    return mul(transform, input_vertex-effect_center);
+}
+
+//-----------------------------------------------------------------------------
+// Shaders
+//-----------------------------------------------------------------------------
+float4 PSSwirl(VertexData vtx) : TARGET {
+    VertexData vertex_out;
+    vertex_out.pos = vtx.pos;
+
+    float2 effect_center = coordinates/2.0;
+    float ar_raw = 1. * ViewSize.y / ViewSize.x;
+    float ar = lerp(ar_raw, 1, aspect_ratio);
+
+    vertex_out.uv = 1. * vtx.uv - effect_center;
+    
+    effect_center.x /= ar;
+    vertex_out.uv.x /= ar;
+    
+    float dist = distance(vertex_out.uv, effect_center);
+    if (dist < radius)
+    {
+        float tension_radius = lerp(radius-dist, radius, tension);
+        float percent = (radius-dist) /tension_radius;
+        percent = inverse_angle == 0 ? percent : 1 - percent;
+
+        vertex_out.uv = swirl_transform(percent, vertex_out.uv, effect_center);
+        vertex_out.uv += (2 * effect_center);
+        
+        vertex_out.uv.x *= ar;
+
+        return InputA.Sample(texture_sampler, vertex_out.uv);
+    }
+    else
+    {
+        return InputA.Sample(texture_sampler, vtx.uv );
+    }
+}
+
+technique Swirl
+{
+    pass 
+    {
+        vertex_shader = DefaultVertexShader(vtx);
+        pixel_shader = PSSwirl(vtx);
+    }
+}
```

Comment:
- The file is in the wrong location, `data/effects` contains internal effects for filters/transitions/sources, not examples. Examples are located at `data/examples`, with filter shader examples being at `data/examples/filter`.
- This project uses 4-wide tabs, [as can be seen in the editorconfig file](https://github.com/Xaymar/obs-StreamFX/blob/master/.editorconfig).

Reply:
- Moved the file to the correct location.
- Fixed the tab spacing according to the .editorconfig contents.

Reply:
Document still contains a lot of spaces for indentation. You might have your editor set to produce spaces instead of tabs.